server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -154,13 +154,21 @@ public static class Builder extends FieldMapper.Builder {
         private final Parameter<VectorSimilarity> similarity;
         private final Parameter<IndexOptions> indexOptions = new Parameter<>(
             "index_options",
-            false,
+            true,
             () -> null,
             (n, c, o) -> o == null ? null : parseIndexOptions(n, o),
             m -> toType(m).indexOptions,
             XContentBuilder::field,
             Objects::toString
-        ).setSerializerCheck((id, ic, v) -> v != null);
+        ).setSerializerCheck((id, ic, v) -> v != null)
+            .setMergeValidator(
+                (previous, current, c) -> previous == null
+                    || Objects.equals(previous, current)
+                    || previous.type.equals(VectorIndexType.FLAT.name)
+                    || previous.type.equals(VectorIndexType.HNSW.name) && current.type.equals(VectorIndexType.INT8_HNSW.name)
+                    || previous.type.equals(VectorIndexType.INT8_FLAT.name) && current.type.equals(VectorIndexType.HNSW.name)
```

Comment:
I think this would could simplified if each `IndexOptions` object had a new `canChangeTo` or `updateableTo`. This way it ends up being 

```
        ).setSerializerCheck((id, ic, v) -> v != null)
            .setMergeValidator(
                (previous, current, c) -> previous == null || previous.updateableTo(current) 
```

Reply:
I like your suggestion, thanks Ben ;)